File: crm_frontend/fix2.mjs
import fs from 'fs';

const filepath = 'src/pages/admin/UserManagement.tsx';
let content = fs.readFileSync(filepath, 'utf8');

const stateHookRegex = /const \[form, setForm\] = useState\(\{[\s\S]*?reportsTo: "none",?[\s\n]*\}\);/g;
const fixedStateHook = `const [form, setForm] = useState({
        name: "",
        email: "",
        phone: "",
        password: "",
        roleId: "",
        reportsTo: "none",
    });`;

const setFormRegex = /setForm\(\{[\s\S]*?reportsTo: editingUser.reportsTo \|\| "none",?[\s\n]*\}\);/g;
const fixedSetForm = `setForm({
                name: editingUser.name,
                email: editingUser.email,
                phone: editingUser.phone || "",
                password: "",
                roleId: editingUser.roleId || "",
                reportsTo: editingUser.reportsTo || "none",
            });`;

const emptyFormRegex = /\}\s*else\s*\{\s*setForm\(\{[\s\S]*?reportsTo:\s*"none"\,?\s*\}\);\s*\}/g;
const fixedEmptyForm = `} else {
            setForm({ name: "", email: "", phone: "", password: "",  roleId: "", reportsTo: "none" });
        }`;


content = content.replace(stateHookRegex, fixedStateHook);
content = content.replace(setFormRegex, fixedSetForm);
content = content.replace(emptyFormRegex, fixedEmptyForm);

fs.writeFileSync(filepath, content);

File: crm_frontend/fix2.py
import os
import re

file_path = 'src/pages/admin/UserManagement.tsx'
with open(file_path, 'r') as f:
    content = f.read()

# Remove 'teamType: string;' from User[] and interfaces in file
content = re.sub(r'\W*teamType\??:\s*string;?', '', content)
content = re.sub(r'const TEAM_TYPES.*?\];', '', content, flags=re.DOTALL)

# Fix form state object structure
content = re.sub(r'teamType:\s*[\'"].*?[\'"],?', '', content)

with open(file_path, 'w') as f:
    f.write(content)
